File: java/day01/Solution.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Solution {
    public static void main(String[] args) {
        String inputFile = "src/input.txt";
        List<Integer> list1 = new ArrayList<>();
        List<Integer> list2 = new ArrayList<>();

        try (BufferedReader br = new BufferedReader(new FileReader(inputFile))) {
            String line = "";
            while ((line = br.readLine()) != null) {
                String[] parts = line.trim().split("\\s+");
                list1.add(Integer.parseInt(parts[0]));
                list2.add(Integer.parseInt(parts[1]));
            }
        } catch (IOException e) {
            System.err.println("Error reading file" + e.getMessage());
        }
        Collections.sort(list1);
        Collections.sort(list2);

        int totalDistance = 0;
        for (int i = 0; i < list1.size(); ++i) {
            totalDistance += Math.abs(list1.get(i) - list2.get(i));
        }

        System.out.println("Total distance: " + totalDistance);
    }
}

File: cpp/day01/solution.cpp
#include <algorithm>
#include <cmath>
#include <fstream>
#include <iostream>
#include <vector>

using namespace std;
int main() {
  ifstream inputFile("input.txt");

  if (!inputFile.is_open()) {
    cerr << "Error opening the file!";
    return 1;
  }

  string line = "";
  vector<int> list1;
  vector<int> list2;
  int num1 = 0;
  int num2 = 0;

  while (inputFile >> num1 >> num2) {
    list1.push_back(num1);
    list2.push_back(num2);
  }
  inputFile.close();

  sort(list1.begin(), list1.end());
  sort(list2.begin(), list2.end());

  int totalDistance = 0;
  for (size_t i = 0; i < list1.size(); ++i) {
    totalDistance += abs(list1[i] - list2[i]);
    //totalDistance += abs(static_cast<int>(list1[i] - list2[i]));
  }

  cout << "Total distance: " << totalDistance << endl;
  return 0;
}
